Proxy/Configuration/Tokenize/Text/index.ts

```diff
@@ -1,61 +1,73 @@
-import { Card } from "../../../../Card/index"
-import { Pattern } from "../../.."
+import { Card } from "../../../../Card"
+import { Pattern } from "../../../Pattern"
 import { Card as CardType } from "./Card"
+import { Plain as TextPlain } from "./Plain"
+import { Xml as TextXml } from "./Xml"
 
-export interface Text {
-	type: "text"
-	card: CardType
-	set: (
-		| {
-				find: Pattern
-				value?: string
-		  }
-		| string
-	)[]
-}
+export type Text = TextXml | TextPlain
 
 export namespace Text {
-	export function is(value: Text | any): value is Text {
-		return (
-			typeof value == "object" &&
-			value.type == "text" &&
-			value.card &&
-			(CardType.Base.is(value.card) || CardType.MonthYear.is(value.card) || CardType.Expires.is(value.card)) &&
-			value.set &&
-			Array.isArray(value.set) &&
-			value.set.every((element: string | Record<string, any>) => {
-				return (
-					typeof element == "string" ||
-					(typeof element == "object" &&
-						element.find &&
-						Pattern.is(element.find) &&
-						(element.value == undefined || typeof element.value == "string"))
-				)
-			})
-		)
-	}
-
-	export function extract(configuration: Text, body: string): Card {
+	export function extract(configuration: TextPlain | TextXml, body: string): Card {
+		const plain = TextPlain.is(configuration)
 		let month: Card.Expires.Month | undefined | 0
 		let year: Card.Expires.Year | undefined | 0
 
 		if (CardType.MonthYear.is(configuration.card)) {
-			month = Card.Expires.Month.parse(Pattern.get(body, configuration.card.month))
-			year = Card.Expires.Year.parse(Pattern.get(body, configuration.card.year))
+			month = Card.Expires.Month.parse(
+				plain
+					? Pattern.getInPlain(
+							body,
+							configuration.card.month,
+							configuration.entryDelimiter,
+							configuration.equalsDelimiter
+					  )
+					: Pattern.getInXml(body, configuration.card.month)
```

Comment:
these ifs, and the getInPlain and getInXml methods, which don't share any code at all, are telling you that they shouldn't be in the same class because they don't do the same thing

as an example
```
abstract class Pattern {
    function forConfiguration(configuration: TextXml | TextPlain) : Pattern {
        return TextPlain.is(configuration) ? new PlainPattern(configuration) : new XmlPattern(configuration);
    }

    abstract function extract(body: String) : string;

    abstract function apply(data: any, pattern: Pattern, value: any): string;
}

class PlainPattern extends Pattern {
    // blah blah blah
}

class XmlPattern extends Pattern {
    // blah blah blah
}


// usage
Pattern.forConfiguration(configuration).extract(body)
```

Reply:
probably better now, please check